subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -169,20 +170,33 @@ static int state_ua_pin_wait(void)
 		.message_id = NCT_MSG_ID_STATE_REPORT,
 	};
 
+	/* Prevent duplicate application notifications of user association request
+	 * events. They are unnecessary and caused by various background shadow
+	 * updates.
+	 */
+	if (!jitp_association_wait) {
+		jitp_association_wait = true;
+		add_shadow_info = IS_ENABLED(CONFIG_NRF_CLOUD_SEND_SHADOW_INFO);
+		(void)nct_save_session_state(false);
+	} else {
+		LOG_DBG("Ignoring duplicate STATE_UA_PIN_WAIT");
+		return 0;
+	}
+
 	/* Update the shadow for the current state: STATE_UA_PIN_WAIT */
 	err = nrf_cloud_state_encode(STATE_UA_PIN_WAIT, false, false, &msg.data);
-	if (err) {
+	if (err < 0) {
```

Comment:
What if it changes in the future to also return something else? We don't want to only check for a single error code here -- the caller should not need to know all possible return values.

Reply:
Ah, I see now that I did miss that it can also return `-ENOTSUP`

My main point (which I should have done a better job communicating, sorry), is that I don't think positive values are possible, since all the return statements are either `return 0` or `return -ENOTSUP` or `return -ENOMEM`